### ISSUE TITLE
Flatpak support?

### ISSUE THREAD

**Issue by ilmaisin** [haiwen/seafile-client#847]
Many major Linux distros, including Fedora, Debian and Opensuse, are adopted or in process of adopting Flatpak as a method of installing desktop software. Even Ubuntu [has it in the universe repo](http://packages.ubuntu.com/search?keywords=flatpak), beginning from 16.10.

I think the old, distro-centric software distribution model is quite bureacuratic and requires redundant work that is not always done. For example when I write this, Fedora 25 [does not ship seafile-client](https://apps.fedoraproject.org/packages/seafile-client/overview/) due to the lack of a maintainer.

Another option would be an AppImage, but it does not by itself ensure compatibility with libraries available, requiring manual testing.

**Comment by probonopd** [comment]
[AppImage](http://appimage.org) does not need special infrastructure from the distro and works on most distributions right out-of-the-box.

**Comment by ilmaisin** [comment]
Yes, the lack of the need of special infrastructure is a benefit not to be overlooked. No, AppImage does not gurantee compatibility with any, let alone "most" distributions. I understand you are enthustiastic about your project but no amount of comment spamming will change the facts. AppImage is more like a glorified portable binary.

It can however be enough if there are enough resources to test the cross-distro compatibility.

**Comment by probonopd** [comment]
@ilmaisin AppImage does not "guarantee" anything. It _enables_ upstreams to provide one download that can be used by users of most desktop distributions _if done properly_.

**Comment by ilmaisin** [comment]
@probonopd How does it "enable" it either? Mozilla has done that for years without AppImage.

I think opening another issue would be appopriate for discussion of the AppImage option.

**Comment by ilmaisin** [comment]
Fedora [now has](https://apps.fedoraproject.org/packages/snapd) Snappy available as well as flatpak. It looks good in the terms that software authors in the future can choice either snappy or flatpak depending on their needs.

**Comment by probonopd** [comment]
@ilmaisin it enables application authors to ship one file that contains the application and everything it needs to run that cannot be expected to be part of the target systems.  And it does that _today_, not in the future, on Fedora, Ubuntu, and many other distributions.

Mozilla has done something similar for a long time, and I applaud them for that.

**Comment by FeuRenard** [comment]
I wrote a Flatpak manifest, which successfully builds an app. But I was not able to actually run the built app. The journal says:

> seafile-applet: error while loading shared libraries: libseafile.so.0: cannot open shared object file: No such file or directory

I see this file amongst the installed files, though. Maybe someone with any more experience in building flatpaks can pick up my work: https://gist.github.com/FeuRenard/2500e60535269efae4fc178e2156c841

**Comment by darkdragon-001** [comment]
@FeuRenard Actually the Flatpak built with your manifest is working for me!
When you start, you can also add a command (e.g. `flatpak run com.seafile.seafile-client bash`). Then you can check your environment variables with `env`. Make sure that `LD_LIBRARY_PATH` is set correctly.

Just the app indicator is not shown, but I am running Flatpak on Gnome, so this might be the reason.
